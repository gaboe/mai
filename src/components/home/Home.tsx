import * as React from "react";
import { Header, Container, Button, Icon } from "semantic-ui-react";
import { Link } from "react-router-dom";

const Home: React.SFC = () => (
  <Container text={true}>
    <Header
      as="h1"
      style={{
        fontSize: "4em",
        fontWeight: "normal",
        marginBottom: 0,
        marginTop: "3em"
      }}
    >
      Made with
      <Icon name="heart" />
    </Header>
    <Header
      as="h2"
      content="By Gabo Ečegi"
      style={{
        fontSize: "1.7em",
        fontWeight: "normal",
        marginTop: "1.5em"
      }}
    />
    <Link to="/random-search-dejong-first">
      <Button primary={true} size="huge">
        Start with Random Search
        <Icon name="arrow right" />
      </Button>
    </Link>
  </Container>
);

export { Home };
