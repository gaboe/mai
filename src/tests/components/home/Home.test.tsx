import * as React from "react";
import { Home } from "./../../../components/home/Home";
import * as renderer from "react-test-renderer";
import { HashRouter as Router } from "react-router-dom";

test("Render Home Page", () => {
  const component = renderer.create(
    <Router>
      <Home />
    </Router>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
