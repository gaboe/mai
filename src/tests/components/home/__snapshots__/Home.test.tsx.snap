// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home Page 1`] = `
<div
  className="ui text container"
>
  <h1
    className="ui header"
    style={
      Object {
        "fontSize": "4em",
        "fontWeight": "normal",
        "marginBottom": 0,
        "marginTop": "3em",
      }
    }
  >
    Made with
    <i
      aria-hidden="true"
      className="heart icon"
    />
  </h1>
  <h2
    className="ui header"
    style={
      Object {
        "fontSize": "1.7em",
        "fontWeight": "normal",
        "marginTop": "1.5em",
      }
    }
  >
    By Gabo Eƒçegi
  </h2>
  <a
    href="#/random-search-dejong-first"
    onClick={[Function]}
  >
    <button
      className="ui huge primary button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Start with Random Search
      <i
        aria-hidden="true"
        className="arrow right icon"
      />
    </button>
  </a>
</div>
`;
